scripts/generate_cloudevent_protos.dart

```diff
@@ -26,55 +27,198 @@ void setupLogging() {
+      }
+    }
+  }
+
+  if (!await _moveGeneratedFilesUp()) {
+    return false;
+  }
 
   if (result.exitCode != 0) {
-    logger.severe('Error running protoc: ${result.stderr}');
+    logger.severe(
+        'Protoc error (code ${result.exitCode}):\n${result.stderr}\n${result.stdout}');
+    return false;
+  }
+
+  logger.info('Successfully generated Dart Protobufs.');
+  if (result.stdout.toString().isNotEmpty) {
+    logger.info('Protoc stdout:\n${result.stdout}');
+  }
+
+  return true;
+}
+
+Future<bool> _ensureCleanOutputDir(Directory outputDir) async {
+  if (!outputDir.existsSync()) {
+    await outputDir.create(recursive: true);
   } else {
-    logger.info('Successfully generated Dart Protobufs.');
+    logger.info(
+        'Cleaning contents of existing output directory: ${outputDir.path}');
+    try {
+      for (final entity in outputDir.listSync()) {
+        if (entity is File) {
+          await entity.delete();
+        } else if (entity is Directory) {
+          await entity.delete(recursive: true);
+        }
+      }
+      logger.info('Successfully cleaned contents of ${outputDir.path}');
+    } catch (e, s) {
+      logger.severe(
+          'Failed to clean contents of directory ${outputDir.path}: $e\nStack:\n$s');
+      return false;
+    }
+  }
+
+  return true;
+}
+
+Future<bool> _moveGeneratedFilesUp() async {
+  final sourceDirPath = path_lib.join('lib', 'src', 'google');
+  final targetDirPath = path_lib.join('lib', 'src');
+
+  final sourceDir = Directory(sourceDirPath);
+
+  if (!sourceDir.existsSync()) {
+    logger.info(
+        'Source directory "$sourceDirPath" for moving files does not exist. Skipping move operation.');
+    return false;
+  }
+
+  logger.info('Moving files from "$sourceDirPath" to "$targetDirPath"...');
+  try {
+    await for (final entity
+        in sourceDir.list(recursive: false, followLinks: false)) {
+      final newPath =
+          path_lib.join(targetDirPath, path_lib.basename(entity.path));
+      if (entity is File) {
+        await entity.rename(newPath);
+        logger.info('Moved file: ${entity.path} -> $newPath');
+      } else if (entity is Directory) {
+        // Note: If subdirectories within 'google' also need their contents moved
+        // rather than the directory itself, this logic would need to be more complex.
+        // For now, it moves the directory itself.
+        await entity.rename(newPath);
+        logger.info('Moved directory: ${entity.path} -> $newPath');
+      }
+    }
+    // After moving all contents, delete the source directory
+    await sourceDir.delete();
+    logger.info('Successfully removed directory: "$sourceDirPath"');
+  } catch (e, s) {
+    logger.severe('Error moving files or deleting directory: $e\nStack:\n$s');
+    return false;
   }
+
+  return true;
 }
 
-void main() async {
+// Helper to ensure paths passed to protoc use forward slashes.
+String _toProtocArgPath(String path) {
```

Comment:
I think it'd be cool if we made this an extension of  path_lib. 

```dart
extension ProtocArgPath on path_lib.Context {
  String protocArgPath => path.replaceAll(separator, '/');
}
``` 

It feels like something like this belongs as a member of that library rather than this. But it makes sense for this library to own the implementation.